internal/web/controllers/settings.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/glasskube/glasskube/internal/clicontext"
+	"github.com/glasskube/glasskube/internal/cliutils"
+	"github.com/glasskube/glasskube/internal/web/components/toast"
+	"github.com/glasskube/glasskube/internal/web/cookie"
+	"github.com/glasskube/glasskube/internal/web/responder"
+	"github.com/glasskube/glasskube/internal/web/types"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type settingsPageData struct {
+	types.TemplateContextHolder
+	Repositories    []v1alpha1.PackageRepository
+	AdvancedOptions bool
+}
+
+func GetSettings(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pkgClient := clicontext.PackageClientFromContext(ctx)
+
+	var repos v1alpha1.PackageRepositoryList
+	if err := pkgClient.PackageRepositories().GetAll(ctx, &repos); err != nil {
+		responder.SendToast(w, toast.WithErr(fmt.Errorf("failed to fetch repositories: %w", err)))
+		return
+	}
+
+	advancedOptions, err := cookie.GetAdvancedOptionsFromCookie(r)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to get advanced options from cookie: %v\n", err)
+	}
+
+	responder.SendPage(w, r, "pages/settings",
+		responder.ContextualizedTemplate(&settingsPageData{
+			Repositories:    repos.Items,
+			AdvancedOptions: advancedOptions,
+		}))
+}
+
+func PostSettings(w http.ResponseWriter, r *http.Request) {
+	formVal := r.FormValue(cookie.AdvancedOptionsKey)
+	cookie.SetAdvancedOptionsCookie(w, formVal == "on")
+}
+
+type repositoryPageData struct {
+	types.TemplateContextHolder
+	Repository v1alpha1.PackageRepository
+}
+
+func GetRepository(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pkgClient := clicontext.PackageClientFromContext(ctx)
+
+	repoName := r.PathValue("repoName")
+	var repo v1alpha1.PackageRepository
+	if err := pkgClient.PackageRepositories().Get(ctx, repoName, &repo); err != nil {
+		responder.SendToast(w, toast.WithErr(fmt.Errorf("failed to fetch repositories: %w", err)))
+		return
+	}
+	responder.SendPage(w, r, "pages/repository",
+		responder.ContextualizedTemplate(&repositoryPageData{
+			Repository: repo,
+		}))
+}
+
+func PostRepository(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pkgClient := clicontext.PackageClientFromContext(ctx)
+
+	repoName := r.PathValue("repoName")
+	repoUrl := r.FormValue("url")
+	checkDefault := r.FormValue("default")
+	opts := v1.UpdateOptions{}
+	var repo v1alpha1.PackageRepository
+	var defaultRepo *v1alpha1.PackageRepository
+	var err error
+
+	if err := pkgClient.PackageRepositories().Get(ctx, repoName, &repo); err != nil {
+		responder.SendToast(w, toast.WithErr(fmt.Errorf("failed to fetch repositories: %w", err)))
+		return
+	}
+
+	if repoUrl != "" {
+		if _, err := url.ParseRequestURI(repoUrl); err != nil {
+			responder.SendToast(w, toast.WithErr(fmt.Errorf("use a valid URL for the package repository (got %v)", err)))
+			return
+		}
+		repo.Spec.Url = repoUrl
+	}
+
+	repo.Spec.Auth = nil
+
+	if checkDefault == "on" {
+		// TODO "cliutils" in the server??
```

Comment:
The "cli" part in `cliutils` means that it should not be used by the package operator.
Since the server is part of the "cli" (aka client application) this is OK. 

But I agree that the naming is kind of ambiguous here, since we have two distinctions:
1. CLI <=> operator
2. CLI <=> server